### PR TITLE
build: Use goreman for starting locally.

### PR DESCRIPTION
**Pull request by alexec**
Fixes: #2965 

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to USERS.md.

# Changes

`argo-server` and `workflow-controller` and started using `goreman` same as Argo CD, which foreground them and logs which prepend the app name:

goreman-logs

We label resources and can prune resources not needed. 

This can be merged as soon as green.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -56,16 +56,16 @@ commands:
...
-          name: Preemptively pull images
-          command: make pull-build-images test-images
-          background: true
       - run:
           name: Install and start K3S v1.0.1
           command: curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.0.1 INSTALL_K3S_EXEC=--docker K3S_KUBECONFIG_MODE=644 sh -
           background: true
       - restore_go_cache
       - install_golang
+      - run:
```
Comment (alexec) [inline .circleci/config.yml:65]: fix regular failing builds (due to Golang not being installed)

File: Makefile [hunk 2]
```diff
@@ -310,32 +317,17 @@ $(VERSION_FILE):
...
-ifeq ($(DB),postgres)
-	kubectl -n argo apply -f dist/postgres.yaml
-else
-ifeq ($(DB),mysql)
-	kubectl -n argo apply -f dist/mysql.yaml
-else
-	kubectl -n argo apply -f dist/no-db.yaml
-endif
-endif
+	kubectl -n argo apply -l app.kubernetes.io/part-of=argo --prune --force -f dist/$(DB).yaml
```
Comment (alexec) [inline Makefile:330]: this will delete mysql if you run postgres and vice-versa, but not delete workflows

Comment (simster7) [reply Makefile:330]: Great, thanks for fitting my use case.

File: manifests/quick-start-mysql.yaml [hunk 3]
```diff
@@ -2,6 +2,8 @@
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
+  labels:
+    app.kubernetes.io/part-of: argo
```
Comment (simster7) [inline manifests/quick-start-mysql.yaml:6]: Good add

**Comment by simster7** [comment]
Linked #2965 to this

**Comment by alexec** [comment]
@simster7 thank you

**Pull request merged**